plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/command/edit/EditCommandsViewImpl.java

```diff
@@ -440,7 +455,17 @@ public void onNameKeyUp(KeyUpEvent event) {
 
     @UiHandler("filterInputField")
     public void onFilterKeyUp(KeyUpEvent event) {
-        renderCategoriesList(categories);
+        if (!filterTextValue.equals(filterInputField.getText())) {
+            filterTextValue = filterInputField.getText();
+            renderCategoriesList(categories);
+        }
+    }
+
+    @Override
+    protected void onEnterClicked() {
+        if (saveButton.isEnabled()) {
+            delegate.onSaveClicked();
+        }
     }
 
     interface EditConfigurationsViewImplUiBinder extends UiBinder<Widget, EditCommandsViewImpl> {
```

Comment:
UI-binder interface should be renamed to **EditCommandsViewImplUiBinder** in order to be more related with view class.